src/rpc/rawtransaction.cpp

```diff
@@ -614,8 +620,23 @@ static RPCHelpMan combinerawtransaction()
         }
     }
 
-    if (txVariants.empty()) {
-        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Missing transactions");
+    { // Test Tx relation for mergeability. Strip scriptSigs and scriptWitnesses to facilitate txId comparison
+        std::vector<CMutableTransaction> tx_variants_copy(txVariants);
+        Txid txid{};
+        for (unsigned int k{0}; k < tx_variants_copy.size(); ++k) {
+            // Remove all scriptSigs and scriptWitnesses from inputs
+            for (CTxIn& input : tx_variants_copy[k].vin) {
+                input.scriptSig.clear();
+                input.scriptWitness.SetNull();
+            }
+            if (k == 0) {
+                txid = tx_variants_copy[k].GetHash();
+            } else {
+                if (txid != tx_variants_copy[k].GetHash()) {
```

Comment:
In a516fe8ada1880f30c6e4553677a9a2580a17cb0 "rpc: combinerawtransaction now rejects unmergeable transactions"

nit: can be a single line `else if`